packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -527,16 +607,246 @@ namespace Ifpack2 {
       struct ToBuffer {};
       struct ToMultiVector {};
 
-      // for cuda, kernel launch should be public too.
+      AsyncableImport (const Teuchos::RCP<const tpetra_map_type>& src_map,
+                       const Teuchos::RCP<const tpetra_map_type>& tgt_map, 
+                       const local_ordinal_type blocksize_,
+                       const local_ordinal_type_1d_view dm2cm_) {
+        blocksize = blocksize_;
+        dm2cm = dm2cm_;
+        
+#ifdef HAVE_IFPACK2_MPI
+        const auto mpi_comm = Teuchos::rcp_dynamic_cast<const Teuchos::MpiComm<int> >(tgt_map->getComm());
+        TEUCHOS_ASSERT(!mpi_comm.is_null());
+        comm = *mpi_comm->getRawMpiComm();
```

Comment:
For future reference: We have a function already for extracting the raw `MPI_Comm`: https://github.com/trilinos/Trilinos/blob/master/packages/tpetra/core/src/Tpetra_Details_extractMpiCommFromTeuchos.hpp

Reply:
done.